video/vaapi.c

```diff
@@ -296,6 +263,38 @@ static const struct va_native_display disp_drm = {
 };
 #endif
 
+#if HAVE_VAAPI_X11
+#include <X11/Xlib.h>
+#include <va/va_x11.h>
+
+static void x11_destroy(void *native_ctx)
+{
+    XCloseDisplay(native_ctx);
+}
+
+static void x11_create(VADisplay **out_display, void **out_native_ctx,
+                       const char *path)
+{
+    void *native_display = XOpenDisplay(NULL);
+    if (!native_display)
+        return;
+    *out_display = vaGetDisplay(native_display);
+    if (*out_display) {
+        *out_native_ctx = native_display;
+    } else {
+        XCloseDisplay(native_display);
+#if HAVE_VAAPI_DRM
+        drm_create(out_display, out_native_ctx, path);
```

Comment:
isn't this just repeating the same call from the `va_native_display` array that must have already failed to reach here?

Reply:
Hmm, you're right, will drop that.